twisted/twisted/plugins/tribler_plugin.py

```diff
@@ -55,7 +59,20 @@ def signal_handler(sig, _):
         msg("Starting Tribler")
 
         config = SessionStartupConfig()
-        config.set_http_api_enabled(True)
+
+        if options["statedir"]:
+            config.set_state_dir(options["statedir"])
+
+        if options["restapi"] != 0:
+            config.set_http_api_enabled(True)
+            config.set_http_api_port(options["restapi"])
+
+        if options["dispersy"] != -1:
+            config.set_dispersy_port(options["dispersy"])
```

Comment:
If you set it to any other negative number dispersy will break, I would control for basic argument sanity.